src/statistics/statistics.c

```diff
@@ -242,23 +259,24 @@ static char *format_size(double size) {
 		divisor = 1024 * 1024 * 1024;
 	}
 
-	sprintf(size_str, fmt, size / divisor);
+	snprintf(size_str, 32, fmt, size / divisor);
 
 	return size_str;
 }
 
 
 static inline void statistics_flush_gvt_buffer(void) {
 	FILE *f;
-
-	f = get_file(STAT_PER_THREAD, GVT_STAT);
-	fprintf(f, "%s", gvt_buffer.buff);
-	fflush(f);
-	gvt_buffer.pos = 0;
-	gvt_buffer.buff[0] = '\0';
+	unsigned i, len;
+	f = thread_files[STAT_FILE_T_GVT][local_tid];
+	len = gvt_buf.pos;
+	for(i = 0; i < len; ++i){
+		fprintf(f, " %15lf    %15lf    %15u    %15u\n",
```

Comment:
Watch out! While your implementation is nicer, the string buffer was done for performance reasons. Indeed, in the old code, this entailed a single interaction with the stdlib and a single interaction with the VFS to dump the content. Since this file can become very large (we reached GBs), letting the kernel optimize a single write could be more efficient.
Do you have any clue on whether this loop of printfs could affect performance?